### ISSUE TITLE
Bug in icon for enclosures

### ISSUE THREAD

**Issue by Toflar** [contao/core#8275]
The function `Controller::addEnclosuresToTemplate()` does not consider own icons for assets.

The code does a hardcoded
```php
'icon'      => TL_ASSETS_URL . 'assets/contao/images/' . $objFile->icon,
```

which renders the support for `$GLOBALS['TL_MIME']` useless. Because when I do something like this

```php
$GLOBALS['TL_MIME']['pdf'][1] = 'files/css/images/mime-pdf.png';
```

it will result in `assets/contao/images/files/css/images/mime-pdf.png`.
I consider this a bug because everywhere else where we use the icon of a file, we use `Image::getHtml()` which considers own paths like this:

```php
if (strpos($src, '/') === false)
{
    if (strncmp($src, 'icon', 4) === 0)
    {
         $static = TL_ASSETS_URL;
         $src = 'assets/contao/images/' . $src;
    }
    // etc.
```

**Comment by Toflar** [comment]
Same applies for `ContentMedia` by the way:

```php
$return .= '<li><img src="' . TL_ASSETS_URL . 'assets/contao/images/' . $objFile->icon . '" width="18" height="18" alt="" class="mime_icon"> <span>' . $objFile->name . '</span> <span class="size">(' . $this->getReadableSize($objFile->size) . ')</span></li>';
```

**Comment by leofeyer** [comment]
Hm, turns out this is not really a bug – rather a case which we have forgotten to think of. By default, there are not paths in `$GLOBALS['TL_MIME']`, which of course does not mean that there should never be any. But supporting paths requires quite a few changes and is a new feature IMHO, isn't it?

@contao/developers /cc

**Comment by Toflar** [comment]
Imho it's wrong to put stuff into 'assets/contao/images'. This has never been a folder I considered to be update-safe. So if I cannot put icons outside of this folder, how do I apply this change update-safe?

**Comment by discordier** [comment]
I am tempted to be with @Toflar on this one.
In fact, it is a new feature as pointed out by @leofeyer but (and a big but here) it is a feature which corrects an initial misconception in a feature that is rendering the original feature pretty useless and therefore IMO qualifies as bug.

**Comment by leofeyer** [comment]
> and a big but here

I don't think that this means what you indented to say :smile:

**Comment by leofeyer** [comment]
Fixed in a76e7910dca4c42dd83e4d792a896f8e81f53eca

**Comment by discordier** [comment]
> I don't think that this is an english phrase :smile:

Yeah, you are right. I simply lacked a better phrase. :grinning:
